client/state/data-layer/wpcom/concierge/schedules/shifts/from-api.js

```diff
@@ -0,0 +1,18 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { makeParser } from 'state/data-layer/wpcom-http/utils';
+import responseSchema from './schema';
+
+const transformShift = shift => ( {
```

Comment:
@dmsnell so we are using now a simplified version of [Data Transfer Object](https://martinfowler.com/eaaCatalog/dataTransferObject.html) pattern. I'm happy to see it as a part of the data layer API. It makes Redux store resilient to changes to the API. Now, you have only one place to updated if new data structure is sent from the server. What are next steps, Flor or TypeScript? 😄

Reply:
> we are using now a simplified version of Data Transfer Object pattern

I guess? don't really see too much relationship to that after reading the linked article other than the fact that we are sending serialized data across the line. `fromApi` is all about recognizing and adhering to well-defined interfaces. our servers tend to send data in a format which is convenient for them while Calypso has different needs. if you're talking about sending the list of shifts here I don't know - seems pretty standard.

this is about the fifth iteration of my attempts at finding ways to build explicit data models in Calypso; TypeScript was an earlier and rejected attempt. JavaScript is really our nemesis here - we're feeling the burden of having a poor type system and trying to hack one on. TypeScript or OCaml or _any_ other language with strong static typing would be a big boon to all this state code.